### PR TITLE
Joss Paper

### PR DESCRIPTION
**Pull request by DanRyanIrish**
### Description

This PR provides a draft ndcube paper for submission to JOSS.  This currently only aims to work out the text and citations.  An author list will be considered subsequently.

To Do
- [ ] Make NDCollection figure
- [ ] Sort out references

### CODE REVIEW

File: joss_paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    orcid: 0000-0001-8661-3825
+    affiliation: 1 # (Multiple affiliations must be quoted)
+  - name: Stuart Mumford^[co-first author] # note this makes a footnote saying 'co-first author'
+    affiliation: 2
+  - name: Author with no affiliation^[corresponding author]
+    affiliation: 3
+affiliations:
+ - name: University of Applied Sciences Northwest Switzerland
+   index: 1
+ - name: University of Sheffield
```
Comment (Cadair) [inline joss_paper/paper.md:17]:
```suggestion
 - name: Aperio Software Ltd
```

File: joss_paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+date: 11 December 2021
+bibliography: paper.bib
+
+---
+
+# Summary
+
+ndcube is a free, open-source, community-developed Python package whose purpose
+is to faciliate the manipulation and visualization of astronomical data.
+It does this by linking the data and coordinates in single data objects.
```
Comment (Cadair) [inline joss_paper/paper.md:28]: "single data object**s**" reads weird.

File: joss_paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+These objects can be manipulated via array-like slicing operations which modify
+the data and coordinates simultaneously.
+They also allow coordinate transformations to be performed with reference to the
+shape of the data and enable coordinate-aware visualizations.
+This data-coordinate coupling allows users to analyze their data more
+easily and accurately, thus helping to boost their scientific output.
```
Comment (Cadair) [inline joss_paper/paper.md:34]: I think we should probably mention, and reference, Astropy's APE 14 here, that in reality is what makes us unique among the wider scientific python ecosystem. https://zenodo.org/record/1188875

File: joss_paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+# Statement of Need
+
+N-dimensional data sets are common in all areas of science and beyond.
+For example, a series of images taken sequentially with a CCD camera can be stored
+as a single 3-D array with two spatial axes and one temporal axis.
+The value in each array-element can represent the reading in a pixel at a given time.
+In astronomy, the relationship between the array element and the location and time
+in the Universe being observed is often represented by the World Coordinate System (WCS) framework.
+WCS’s ability to handle many different physical types (e.g. spatial, temporal, spectral, etc.)
+and projections (e.g. RA and Dec., helioprojective latitude and longitude, etc.)
```
Comment (Cadair) [inline joss_paper/paper.md:61]: We should mention this is a *functional* framework where the transformation is modelled and then evaluated on demand, vs lookup tables.

File: joss_paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+However, they are not suited to treating data and coordinates in a combined way.
+This is the purpose of ndcube.
+
+The fundamental reason to opt for ndcube over its most similar package, xarray,
+is to harness the astronomy-specific World Coordinate System (WCS).
+The data model of xarray centers on the requirements and conventions of the geosciences.
+Although similar to those of astronomy in conception, they are sufficiently different
+in construction to cause significant friction.
+Moreover, utilizing the astropy WCS infrastructure enables us to directly read the most
+common file format in astronomy, FITS.
```
Comment (Cadair) [inline joss_paper/paper.md:75]: Do we need this?!

File: joss_paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+# Data Objects: The Pillars of ndcube
+
+At the time of writing (ndcube v2.0), ndcube provides three primary data objects for
+manipulating astronomical data: NDCube, NDCubeSequence, and NDCollection.
```
Comment (Cadair) [inline joss_paper/paper.md:80]:
```suggestion
manipulating astronomical data: `NDCube`, `NDCubeSequence`, and `NDCollection`.
```

**Review by Cadair — COMMENTED** [review]
When we did this with sunpy core, we had everyone with a commit on the author list.

File: joss_paper/paper.md (hunk 2 above)
Comment (DanRyanIrish) [reply joss_paper/paper.md:28]: unified data objects?

File: joss_paper/paper.md (hunk 5 above)
Comment (DanRyanIrish) [reply joss_paper/paper.md:75]: No!  I've removed the whole paragraph locally.

File: joss_paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+For a quick reference, the following citation commands can be used:
+- `@author:2001`  ->  "Author et al. (2001)"
+- `[@author:2001]` -> "(Author et al., 2001)"
+- `[@author1:2001; @author2:2001]` -> "(Author1 et al., 2001; Author2 et al., 2002)"
+
+# Acknowledgements
+
+We acknowledge support for ndcube from NASA's Heliophysics Data Environment Enhancement
+program and the Daniel K. Inouye Solar Telescope.
+We also acknowledge the SunPy and Python in Heliophysics communities for their support.
```
Comment (DanRyanIrish) [inline joss_paper/paper.md:170]:
```suggestion
The ndcube developers would like to thank the Google Summer of Code program,
NASA's Heliophysics Data Environment Enhancement program,
and the Daniel K. Inouye Solar Telescope for their support.
They would also like to thank the SunPy and Python in Heliophysics communities for their support.
```

**Pull request closed without merging**